survey/survey_app/permissions.py

```diff
@@ -0,0 +1,10 @@
+from rest_framework.permissions import BasePermission
+from .models import Survey
+
+class CanUpdateSurveyPermission(BasePermission):
+    def has_permission(self, request, view):
+        survey_id = view.kwargs.get('survey_id')
+        survey = Survey.objects.get(pk=survey_id)
```

Comment:
@newDaria we often use `try ... exception` when using .get() because there's scenario that the Survey object does not exist.

Reply:
```
from rest_framework.permissions import BasePermission
from django.shortcuts import get_object_or_404
from survey_app.models import Survey

class CanUpdateSurveyPermission(BasePermission):
    def has_permission(self, request, view):
        survey_id = view.kwargs.get('survey_id')
        try:
            survey = get_object_or_404(Survey, pk=survey_id)
        except Survey.DoesNotExist:
            # If the Survey object does not exist, return False to deny permission
            return False

        # Check if the current user is the creator of the survey
        return request.user == survey.creator
```